### ISSUE TITLE
Sync interval settings are ignored after manual sync

### ISSUE THREAD

**Issue by alexdevero** [laurent22/joplin#1489]
After manual sync app automatically syncs notes when some note is changed, even though sync interval is set to "disabled".

### Operating system

- Windows 10

### Application

- Desktop

**Comment by CalebJohn** [comment]
Isn't this a duplicate of #391 which was closed as a duplicate of #310? You should try bumping that issue (which you created) rather than creating a new issue.

**Comment by alexdevero** [comment]
I disagree. This is not about auto-sync and auto-save firing automatically. This occurs only after you sync automatically, not otherwise.

**Comment by CalebJohn** [comment]
I'm sorry I still don't understand how this issue is different. To me in both the issue is that Joplin automatically syncs after not changes. Is this not the case?

**Comment by innocuo** [comment]
The app syncs 30 seconds after you stop editing a note, regardless of what "sync interval" is set to, and regardless of if you previously synced manually or not. 

It isn't a full sync, it only uploads changes, but doesn't download changes from other notes. So "sync interval disabled" isn't the same as sync disabled.

@laurent22  commented in #310: 

> there are two types of sync:
>**Regular sync,** which you can configure in the Option screen. It sends all the latest changes, delete notes as needed and download the latest changes.
>**Sync after save.** It's a light sync operation that only sends the latest changes and it cannot be configured at the moment.
[(link to full comment)](https://github.com/laurent22/joplin/issues/310#issuecomment-381627827)

Aren't you referring to the "sync after save" type? If so, as this is done on purpose, you probably should explain why you think this feature should change.

**Comment by alexdevero** [comment]
@innocuo thanks for explanation. I am probably referring to the "sync after save".  One reason for change is when you are offline, or firewall block the app. Joplin will show error message every time it tries to connect (after every save). In this case trying to sync the data is not necessary. I know that I am offline, or firewall blocks the app.

**Comment by laurent22** [comment]
Normally I've set it up so that it doesn't display network errors. What's the exact message you're having?

**Comment by alexdevero** [comment]
This:
```
Dokončeno: 06/05/2019 05:12
Poslední chyba: FetchError: request to https://graph.microsoft.com/v1.0/drive/root:/Apps/Joplin/.sync?select=name%2Cfile%2Cfolder%2CfileSystemInfo%2CparentReference failed, reason: connect EACCES 40.126.9.50:443
```

**Comment by laurent22** [comment]
I was about to add a check to prevent it from displaying an message for this error, but in the end I think this error should be displayed. If a user has their firewall misconfigured and Joplin cannot sync, we can't really fail silently, the user needs to know about this.

I guess the wider issue in your case is that an executable is created in a random location when starting. Is it using the PortableApps package (PAF package) or the JoplinPortable.exe executable? Can't you whitelist the main executable in the firewall and that should take care of the executables it might create?

**Comment by alexdevero** [comment]
The executable is `C:\users\foo\appdata\local\temp\nsp1bb0.tmp\app\joplin.exe`. I am using Windows Firewall Control and this program allows to set rules only for specific paths. Creating rule for this temporary executable then works only until the next start of Joplin.

**Comment by innocuo** [comment]
Maybe: what if the app stops trying to sync if it fails X consecutive times? specially if it's due to network errors. Syncing manually or restarting the app would clear this flag.

**Comment by alexdevero** [comment]
@innocuo I like Your idea. This could work. Determining a limit to how many times the app will try to sync. Then, if all fail, the app would not try to sync again. This limit could be reset after every successful manual/automatic sync, to make sure the app will not stop the sync forever.

What do you think @laurent22?
